### PR TITLE
bugfix : fix rpcContext.getClientRMHolderMap() npe

### PR DESCRIPTION
**Pull request by doubleDimple**
- [ ] I have registered the PR [changes](https://github.com/seata/seata/tree/develop/changes).

### Ⅱ. Does this pull request fix one issue?
fixes https://github.com/seata/seata/issues/4929

### CODE REVIEW

File: core/src/main/java/io/seata/core/rpc/netty/ChannelManager.java [hunk 1]
```diff
@@ -240,10 +240,12 @@ public static Channel getSameClientChannel(Channel channel) {
             ConcurrentMap<Integer, RpcContext> clientRpcMap = TM_CHANNELS.get(clientIdentified);
             return getChannelFromSameClientMap(clientRpcMap, clientPort);
         } else if (clientRole == NettyPoolKey.TransactionRole.RMROLE) {
-            for (Map<Integer, RpcContext> clientRmMap : rpcContext.getClientRMHolderMap().values()) {
-                Channel sameClientChannel = getChannelFromSameClientMap(clientRmMap, clientPort);
-                if (sameClientChannel != null) {
-                    return sameClientChannel;
+            if (CollectionUtils.isNotEmpty(rpcContext.getClientRMHolderMap())) {
```
Comment (a364176773) [inline core/src/main/java/io/seata/core/rpc/netty/ChannelManager.java:243]: get再赋值,这样下面不需要再getClientRMHolderMap一次

Comment (doubleDimple) [reply core/src/main/java/io/seata/core/rpc/netty/ChannelManager.java]: get

**Review by a364176773 — APPROVED** [review]
LGTM

**Review by slievrly — APPROVED** [review]
LGTM

**Comment by slievrly** [comment]
@doubleDimple add changes in develop.md .

**Pull request merged**